src/rewriter.ts

```diff
@@ -31,12 +31,12 @@ export class Rewriter {
           if (!resolver) {
             throw new Error(`Could not locate resolver for value ${value}`);
           }
-          const innerValue = matchResults ? matchResults[2] : value;
-          let innerResult = await resolver(innerValue, this.config);
-          if (!innerResult) {
-            throw new Error(`Resolver ${resolverName} didn't return any value`);
+          const argument = matchResults ? matchResults[2] : value;
+          let resolved = await resolver(argument, this.config);
+          const rewrittenValue = await this.rewriteValue(resolved);
+          if (rewrittenValue) {
```

Comment:
A little weirded out about this line.  I want to make sure that this will always end up being a string, otherwise this could run into issues.

Will probably do one more push before merging.